### ISSUE TITLE
Add extension for describing generics

### ISSUE THREAD

**Issue by rsuter** [NJsonSchema/NJsonSchema#23]
...

**Comment by grovesNL** [comment]
How should multiple generic parameters be handled with that schema definition?

**Comment by rsuter** [comment]
Yes, you would have multiple entries in the "x-generics" object.

**Comment by rsuter** [comment]
I've updated the sample

**Comment by mirata** [comment]
@RicoSuter I think this would be a really good thing to add actually. Happy to help out if I can commit some time. 
I had a very complicated object I needed to deal with on the typescript side with multiple generics, and it really doesn't work well. I don't think it matters that this concept would only work within the NSwag toolchain - people would definitely make use of it.

**Comment by john-larson** [comment]
Is there any progress on this issue?

**Comment by robertsundstrom** [comment]
> Is there any progress on this issue?

I am wondering the same.

**Comment by RicoSuter** [comment]
> Is there any progress on this issue?

The answer is no there is no progress :-)... not planned as it is very custom to the NJsonSchema/NSwag toolchain - main recommendation is to avoid generics because other tools will not understand it.

**Comment by binginsin** [comment]
Was looking into generics and JSON schema recently and found this:
https://gist.github.com/kevinswiber/4ee078f7bee175852587aca738377168#file-page-schema-json-L32

Apparently generics can be supported using $dynamicRef, $dynamicAnchor and multiple schemas.

I'm not entirely sure how difficult this would be to implement, but I believe it may be possible to add generic support without going against the specification of OpenAPI, even if other generators don't have this implemented.

Interested to hear others' thoughts.

**Comment by antonGritsenko** [comment]
>  because other tools will not understand it.
IMHO this is bad argument, since each and very OpenAPI tool works in a very different way, and none of them fully support all aspects of the OpenAPI on different platform. NJsonSchema and NSwag are from .net world, so they should support generics, because its native of the platform.

**Comment by JohnGalt1717** [comment]
How is this still an issue in 2023 9 years after the first request for generics in the OpenAPI spec that basically all languages support or could just use object?

You should be able to define Page<TResult> as a schema type, and to be valid, everywhere it's referenced should have to provide the ref/concrete type to TResult.

And that should be it.  NSwag shouldn't have to be doing this, this should just be part of the OpenApi spec.  But like Enums that it still has a mess of, this still isn't supported.

This is literally why people create new competing standards: Basics broken.
